### DISCUSSION TITLE
How do I type a stor if _GettersTree is marked as for internal use only?

### DISCUSSION THREAD

**Discussion by Azema4ka** [vuejs/pinia#751] [Help and Questions] [answered]
# The Problem
I used `version 2.0.0-rc.15`, and there the `GettersTree` type was not marked as for internal use only. After upgrading to `version 2.0.0` the `GettersTree` type became `_GettersTree`, how should I better type the store?
# Example of using the store with `version 2.0.0`
```js
interface State extends StateTree {
  list: Apartment[];
}

interface Getters extends GettersTree<State> {
  getById(state: State): (id: string) => Apartment | undefined;
}

export const useApartmentsStore = defineStore<"apartments-pinia", State, Getters>(
  "apartments-pinia",
  {
    state(): State {
      return {
        list: [],
      };
    },
    getters: {
      getById: (state) => (id) => state.list.find((apartment) => apartment.id === id),
    },
  }
);
```

**Comment by BenShelton** [comment, marked as answer]
TS can infer the types without you having to specify them all. This should be all you need:

```ts
interface State {
  list: Apartment[];
}

export const useApartmentsStore = defineStore(
  "apartments-pinia",
  {
    state: (): State => ({
      list: [],
    }),
    getters: {
      getById: (state) => (id: string) => state.list.find((apartment) => apartment.id === id),
    },
  }
);
```

I also converted the state function to an arrow function as it improves type safety for some, see https://pinia.esm.dev/core-concepts/state.html#state

**Reply by Azema4ka** [reply]
> TS can infer the types without you having to specify them all. This should be all you need:
> 
> ```ts
> interface State {
>   list: Apartment[];
> }
> 
> export const useApartmentsStore = defineStore(
>   "apartments-pinia",
>   {
>     state: (): State => ({
>       list: [],
>     }),
>     getters: {
>       getById: (state) => (id: string) => state.list.find((apartment) => apartment.id === id),
>     },
>   }
> );
> ```
> 
> I also converted the state function to an arrow function as it improves type safety for some, see https://pinia.esm.dev/core-concepts/state.html#state

Thank you for your time, your reply was very helpful to me 😘

**Reply by BenShelton** [reply]
Happy to help, do you mind marking this as the correct answer so others can benefit too? Thanks 😃

**Reply by Azema4ka** [reply]
> Happy to help, do you mind marking this as the correct answer so others can benefit too? Thanks 😃

Yes, of course.
